tests/test_builder_containerbuild.py

```diff
@@ -1206,40 +1209,65 @@ def test_build_nvr_exists(self, tmpdir, orchestrator, df_release, param_release,
             task.handler(src['src'], 'target', opts=additional_args)
         assert 'already exists' in str(exc_info.value)
 
-    @pytest.mark.parametrize(('create_args', 'cause'), (
-        ({'koji_build_id': 12345}, 'doesnt exist'),
-        ({'koji_build_nvr': 'build_nvr'}, 'doesnt exist'),
-        ({'koji_build_nvr': 'build_nvr', 'koji_build_id': 12345}, 'doesnt exist'),
-        ({'koji_build_nvr': 'build_nvr', 'koji_build_id': 12345}, 'mismatch'),
+    @pytest.mark.parametrize(('create_args', 'build_type', 'cause'), (
+        ({'koji_build_id': 12345},
+         'image', 'doesnt exist'),
+        ({'koji_build_nvr': 'image_build-1-1'},
+         'image', 'doesnt exist'),
+        ({'koji_build_nvr': 'image_build-1-1', 'koji_build_id': 12345},
+         'image', 'doesnt exist'),
+        ({'koji_build_nvr': 'image_build-1-1', 'koji_build_id': 12345},
+         'image', 'mismatch'),
+        ({'koji_build_nvr': 'rpm_build-1-1', 'koji_build_id': 12345},
+         'rpm', 'wrong type'),
+        ({'koji_build_nvr': 'image_build-source-1-1', 'koji_build_id': 12345},
+         'image', 'source build'),
     ))
-    def test_source_build_info(self, create_args, cause):
+    def test_source_build_info(self, create_args, build_type, cause):
         koji_task_id = 123
         last_event_id = 456
         koji_build_id = 999
         different_build_id = 54321
 
+        provided_nvr = create_args.get('koji_build_nvr')
+        provided_id = create_args.get('koji_build_id')
+        provided_name = None
+        if provided_nvr:
+            provided_name = provided_nvr.rsplit('-', 2)[0]
+
         session = self._mock_session(last_event_id, koji_task_id)
+        (session).should_receive('getBuildType').and_return({build_type: {}})
```

Comment:
Could we add a test with multiple build types being returned (this is the case for operators)

Reply:
yes